### ISSUE TITLE
Unable to deploy QA branch

### ISSUE THREAD

**Issue by Janell-Huyck** [uclibs/aaec#212]
The QA branch currently cannot be deployed.  The last deployable PR was # [baac32cae5fb9b5fae56ee31e08b78f13bc4e0d1](https://github.com/uclibs/aaec/commit/baac32cae5fb9b5fae56ee31e08b78f13bc4e0d1) on Sept 11, description: Merge pull request #191 from uclibs/dependabot/npm_and_yarn/webpack-cli-4.10.0

All later versions of QA error out when we try to deploy them with capistrano.  The following errors are displayed:
```
00:25 deploy:assets:precompile
      01 RBENV_ROOT=$HOME/.rbenv RBENV_VERSION=3.2.2 $HOME/.rbenv/bin/rbenv exec bundle exec rake assets:precompile
      01 yarn install v1.22.19
      01 [1/4] Resolving packages...
      01 [2/4] Fetching packages...
      01 error sass@1.67.0: The engine "node" is incompatible with this module. Expected version ">=14.0.0". Got "12.22.12"
      01 error Found incompatible module.
      01 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
      01 Compiling...
      01 Compilation failed:
      01 yarn run v1.22.19
      01 info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
      01
      01
      01 error Command "webpack" not found.
      01
#<Thread:0x00000001083c2658 /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/runners/parallel.rb:10 run> terminated with exception (report_on_exception is true):
/Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/runners/parallel.rb:15:in `rescue in block (2 levels) in execute': Exception while executing as apache@libappstest.libraries.uc.edu: rake exit status: 1 (SSHKit::Runner::ExecuteError)
rake stdout: yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Compiling...
Compilation failed:
yarn run v1.22.19
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.


error Command "webpack" not found.
rake stderr: error sass@1.67.0: The engine "node" is incompatible with this module. Expected version ">=14.0.0". Got "12.22.12"
error Found incompatible module.

        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/runners/parallel.rb:11:in `block (2 levels) in execute'
/Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/command.rb:97:in `exit_status=': rake exit status: 1 (SSHKit::Command::Failed)
rake stdout: yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Compiling...
Compilation failed:
yarn run v1.22.19
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.


error Command "webpack" not found.
rake stderr: error sass@1.67.0: The engine "node" is incompatible with this module. Expected version ">=14.0.0". Got "12.22.12"
error Found incompatible module.

        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/backends/netssh.rb:170:in `execute_command'
        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/backends/abstract.rb:148:in `block in create_command_and_execute'
        from <internal:kernel>:90:in `tap'
        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/backends/abstract.rb:148:in `create_command_and_execute'
        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/backends/abstract.rb:80:in `execute'
        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/capistrano-rails-1.6.3/lib/capistrano/tasks/assets.rake:69:in `block (6 levels) in <top (required)>'
        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/backends/abstract.rb:100:in `with'
        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/capistrano-rails-1.6.3/lib/capistrano/tasks/assets.rake:68:in `block (5 levels) in <top (required)>'
        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/backends/abstract.rb:92:in `within'
        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/capistrano-rails-1.6.3/lib/capistrano/tasks/assets.rake:67:in `block (4 levels) in <top (required)>'
        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/backends/abstract.rb:31:in `instance_exec'
        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/backends/abstract.rb:31:in `run'
        from /Users/huyckjl/.rbenv/versions/3.2.2/lib/ruby/gems/3.2.0/gems/sshkit-1.21.5/lib/sshkit/runners/parallel.rb:12:in `block (2 levels) in execute'
(Backtrace restricted to imported tasks)
cap aborted!
SSHKit::Runner::ExecuteError: Exception while executing as apache@libappstest.libraries.uc.edu: rake exit status: 1
rake stdout: yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Compiling...
Compilation failed:
yarn run v1.22.19
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.


error Command "webpack" not found.
rake stderr: error sass@1.67.0: The engine "node" is incompatible with this module. Expected version ">=14.0.0". Got "12.22.12"
error Found incompatible module.


Caused by:
SSHKit::Command::Failed: rake exit status: 1
rake stdout: yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Compiling...
Compilation failed:
yarn run v1.22.19
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.


error Command "webpack" not found.
rake stderr: error sass@1.67.0: The engine "node" is incompatible with this module. Expected version ">=14.0.0". Got "12.22.12"
error Found incompatible module.

Tasks: TOP => deploy:assets:precompile
(See full trace by running task with --trace)
The deploy has failed with an error: Exception while executing as apache@libappstest.libraries.uc.edu: rake exit status: 1
rake stdout: yarn install v1.22.19
[1/4] Resolving packages...
[2/4] Fetching packages...
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
Compiling...
Compilation failed:
yarn run v1.22.19
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.


error Command "webpack" not found.
rake stderr: error sass@1.67.0: The engine "node" is incompatible with this module. Expected version ">=14.0.0". Got "12.22.12"
error Found incompatible module.


** DEPLOY FAILED
** Refer to log/capistrano.log for details. Here are the last 20 lines:


  INFO The Gemfile's dependencies are satisfied, skipping installation

 DEBUG [900465aa] Running if test ! -d /opt/webapps/aaec/releases/20230918150525; then echo "Directory does not exist '/opt/webapps/aaec/releases/20230918150525'" 1>&2; false; fi as apache@libappstest.libraries.uc.edu

 DEBUG [900465aa] Command: if test ! -d /opt/webapps/aaec/releases/20230918150525; then echo "Directory does not exist '/opt/webapps/aaec/releases/20230918150525'" 1>&2; false; fi

 DEBUG [900465aa] Finished in 0.186 seconds with exit status 0 (successful).

  INFO [38d44fdb] Running RBENV_ROOT=$HOME/.rbenv RBENV_VERSION=3.2.2 $HOME/.rbenv/bin/rbenv exec bundle exec rake assets:precompile as apache@libappstest.libraries.uc.edu

 DEBUG [38d44fdb] Command: cd /opt/webapps/aaec/releases/20230918150525 && ( export PATH="$PATH:/usr/local/bin" RBENV_ROOT="$HOME/.rbenv" RBENV_VERSION="3.2.2" RAILS_ENV="production" RAILS_GROUPS="" ; RBENV_ROOT=$HOME/.rbenv RBENV_VERSION=3.2.2 $HOME/.rbenv/bin/rbenv exec bundle exec rake assets:precompile )

 DEBUG [38d44fdb]       yarn install v1.22.19

 DEBUG [38d44fdb]       [1/4] Resolving packages...

 DEBUG [38d44fdb]       [2/4] Fetching packages...

 DEBUG [38d44fdb]       error sass@1.67.0: The engine "node" is incompatible with this module. Expected version ">=14.0.0". Got "12.22.12"

 DEBUG [38d44fdb]       error Found incompatible module.

 DEBUG [38d44fdb]       info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.

 DEBUG [38d44fdb]       Compiling...

 DEBUG [38d44fdb]       Compilation failed:

yarn run v1.22.19

info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.





error Command "webpack" not found.

```

**Comment by Janell-Huyck** [comment]
Broken Merges:
29e782e5eb2b5f9e41838c7bd1ffedbd7bd46570,Merge pull request #191 from clibs/dependabot/npm_and_yarn/webpack-cli-4.10.0, on Sept 11

**Comment by Janell-Huyck** [comment]
Merges that Work:

- 0e6a7e55f189b4fc92a55d698e21f5cfff570c6e,Merge pull request #199 from uclibs/fix-fontawesome fixes #161-fontawesome icons
- 5e66d99497d949fd74963c67773dfcfeb8e1a5bd,Merge pull request #192 from uclibs/dependabot/bundler/listen-3.8.0 Bump listen from 3.0.8 to 3.8.0

**Comment by hortongn** [comment]
I've very out of the loop on the deploy problem and you may be past this piece of it but after looking at the errors for the first time just now, I suspect that the webpacker errors are caused by the incompatible node version.  webpacker won't work if node isn't working.

The first error `error sass@1.67.0: The engine "node" is incompatible with this module. Expected version ">=14.0.0". Got "12.22.12"` seems to indicate that a change in SASS is no longer happy with the version of node we're running on the server.  I see that one of the recent dependabot updates was for sass-rails.

**Comment by Janell-Huyck** [comment]
> I've very out of the loop on the deploy problem and you may be past this piece of it but after looking at the errors for the first time just now, I suspect that the webpacker errors are caused by the incompatible node version. webpacker won't work if node isn't working.
> 
> The first error `error sass@1.67.0: The engine "node" is incompatible with this module. Expected version ">=14.0.0". Got "12.22.12"` seems to indicate that a change in SASS is no longer happy with the version of node we're running on the server. I see that one of the recent dependabot updates was for sass-rails.

I agree, Glen.  I believe the server is running 12.22.12 instead of 14+, and that we will not be able to update our dependencies until we find a way to deal with this.

For now, I am focusing on rebuilding what I can out of the PR's that we have made, doing cherry-picking on them.  The end result of that should be an app that can be deployed, has older dependencies, but includes the rest of the changes we have made in the past few weeks.

**Comment by Janell-Huyck** [comment]
@scherztc That's fantastic!  I went ahead and test-deployed it on qa, and it deploys fine.  It runs locally fine.  I'd feel more comfortable if @crowesn or @hortongn took a peek at the PR before we merge it up, but it looks good to me.  Thank you!!

**Comment by Janell-Huyck** [comment]
Fixed by PR #213
